src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -113,27 +109,92 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
                 copyCommandDict['checksum'] = "-K adler32"
 
         copyCommandDict['options'] = ' '.join(unknown)
-
         copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
         copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
 
+        return copyCommandDict
+
+    def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+        """
+        Create gfal-cp command for stageOut
+
+        :sourcePFN: str, PFN of the source file
+        :targetPFN: str, destination PFN
+        :options: str, additional options for gfal-cp
+        :checksums: dict, collect checksums according to the algorithms saved as keys
+        """
+
+
+        copyCommandDict = self.buildCopyCommandDict(sourcePFN, targetPFN, options, checksums)
         copyCommand = self.copyCommand % copyCommandDict
-        result += copyCommand
+        result = "#!/bin/bash\n" + copyCommand
 
         if _CheckExitCodeOption:
             result += """
             EXIT_STATUS=$?
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
-               echo "ERROR: gfal-copy exited with $EXIT_STATUS"
-               echo "Cleaning up failed file:"
-               %s
+                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+                echo "Cleaning up failed file: {remove_command}"
             fi
             exit $EXIT_STATUS
-            """ % self.createRemoveFileCommand(targetPFN)
-
+            """.format(
+                remove_command=self.createRemoveFileCommand(targetPFN)
+            )
+        
         return result
 
+    def createDebuggingCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+        """
+        Debug a failed gfal-cp command for stageOut, without re-running it,
+        providing information on the environment and the certifications
+
+        :sourcePFN: str, PFN of the source file
+        :targetPFN: str, destination PFN
+        :options: str, additional options for gfal-cp
+        :checksums: dict, collect checksums according to the algorithms saved as keys
+        """
+        
+        copyCommandDict = self.buildCopyCommandDict(sourcePFN, targetPFN, options, checksums)
+        copyCommand = self.copyCommand % copyCommandDict
+
+        proxyInfo = None
+        try:
+            stdout, stderr, returnCode = runCommand("voms-proxy-info", timeout=10)
```

Comment:
I am not sure it will work either, as I am not sure it can find the proxy file. We might have to append this command with `-file $X509_USER_PROXY`, but these things need to be tested.